search.go

```diff
@@ -0,0 +1,34 @@
+package registry
+
+import (
+	"fmt"
+)
+
+// SearchParams encapsulates parameters provided to Searchable.Search
+type SearchParams struct {
+	Q             string
+	Limit, Offset int
+}
+
+// Searchable is an interface for supporting search
+type Searchable interface {
+	Search(p SearchParams) ([]Result, error)
+}
+
+// ErrSearchNotSupported is returned for collections that are not searchable
+var ErrSearchNotSupported = fmt.Errorf("search not supported")
+
+// NilSearch is a stub for collections that don't support search
+type NilSearch bool
+
+// Search returns an error indicating that search is not supported
+func (ns NilSearch) Search(p SearchParams) ([]Result, error) {
+	return nil, ErrSearchNotSupported
+}
+
+// Result is the interface that a search result implements
```

Comment:
I'd move this closer to the `Searchable` declaration, keeping the NilSearch implementation below. Generally if interfaces & a base implementation are in the same file all aspects of the interface should be listed first, and the comment on the type should include a note like `foo is a basic implementation of Searchable`